### PR TITLE
OneSignal Integration

### LINKED ISSUE #331: Integrate One Signal ($100)
**Is your feature request related to a problem? Please describe.**
https://documentation.onesignal.com/docs/flutter-sdk-setup

We should be able to send updates, notifications, or anything through push notifications to user, about important updates, deals, etc.

**Describe the solution you'd like**
Implement OneSignal integration + user identification on main.dart

**Additional context**
I spent a few hours trying to set it up but this issue always came: https://github.com/OneSignal/OneSignal-Flutter-SDK/issues?q=Multiple+commands+produce

### PR DESCRIPTION
**Pull request by mdmohsin7**
This PR integrates OneSignal for both Android and iOS.
Fixes #331 
Note: For iOS, the OneSignal Notification Service Extension is only set up for the bundle ID `com.friend-app-with-wearable.ios12`. This means it won't work with the bundle ID `com.friend-app-with-wearable.ios12.dev` because this bundle ID isn't included in the Development Team.

Steps to integrate OneSignal on iOS with xcode 15 or higher:
https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/551#issuecomment-2184159953

Android            |  iOS
:-------------------------:|:-------------------------:
Android  |  iOS

### CODE REVIEW

File: apps/AppWithWearable/lib/main.dart [hunk 1]
```diff
@@ -22,6 +23,15 @@ void main() async {
   await SharedPreferencesUtil.init();
   await MixpanelManager.init();
   await ObjectBoxUtil.init();
+  if (Env.oneSignalAppId != null) {
```
Comment (josancamon19) [inline apps/AppWithWearable/lib/main.dart:26]: Let's login independently of permissions, and let's not ask for permissions when the app opens, currently notification permissions are being asked at home/page.dart

**Review by josancamon19 — CHANGES_REQUESTED** [review]

**Comment by josancamon19** [comment]
App runs seamlessly, but I have not been able to test because of the onesignal account, merging for now as it is, looking forward for the remaining changes :)

File: apps/AppWithWearable/lib/main.dart (hunk 1 above)
Comment (mdmohsin7) [reply apps/AppWithWearable/lib/main.dart:26]: I just now actually made this change and after pushing it, realised that the PR has been merged

**Comment by mdmohsin7** [comment]
> App runs seamlessly, but I have not been able to test because of the onesignal account, merging for now as it is, looking forward for the remaining changes :)

I'm adding your email as an admin in my personal OneSignal account. And also should I send the p8 file through mail or you'll be generating a new on through Apple Developer Console?

File: apps/AppWithWearable/lib/main.dart (hunk 1 above)
Comment (mdmohsin7) [reply apps/AppWithWearable/lib/main.dart:26]: Opened a new PR for this change #347

**Pull request merged**